core/controllers/base_test.py

```diff
@@ -494,3 +494,33 @@ def test_error_response_for_get_request_of_type_json_has_json_format(self):
             response = self.get_json('/fake', expect_errors=True)
             self.assertTrue(isinstance(response, dict))
             self.assertEqual(500, response['code'])
+
+
+class CheckAllHandlersHaveDecorator(test_utils.GenericTestBase):
+    """Tests that all methods in handlers have authentication decorators
+    applied on them."""
+
+    def test_every_method_has_decorator(self):
+        for route in main.URLS:
+            if isinstance(route, tuple):
```

Comment:
URLS = MAPREDUCE_HANDLERS + other handers. MAPREDUCE_HANDLERS are touples. So below check is to handle them.

ok.